### PR TITLE
[JIT] Build SchemaInfo for understanding operator behavior

### PR DESCRIPTION
**Pull request by ttanpcs**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #80359
* #80244
* #80186
* __->__ #80177

- Created SchemaInfo class which publicizes op information through isMutating() and isAliasing() methods.
- Methods tested with sub_ op for mutation, aliasing, and exceptions in test_schema_info.cpp.

Differential Revision: [D37474862](https://our.internmc.facebook.com/intern/diff/D37474862)

### CODE REVIEW

**Comment by goldenxuett** [comment]
@goldenxuett has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37474862).

File: torch/csrc/utils/schema_info.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          (rhs.type == output && rhs.index < schema_.returns().size() &&
+           rhs.index >= 0),
+      "Invalid index for schema.");
+
+  const c10::Argument lhsArg = schema_.arguments()[lhs.index];
+  const c10::Argument rhsArg = schema_.arguments()[rhs.index];
+
+  if (lhsArg.alias_info() && lhsArg.alias_info()->isWildcardAfter() &&
+      rhsArg.alias_info() && rhsArg.alias_info()->isWildcardAfter()) {
+    if (lhsArg.type()->kind() == rhsArg.type()->kind()) {
```
Comment (davidberard98) [inline torch/csrc/utils/schema_info.cpp:44]: https://github.com/pytorch/pytorch/blob/4ee29d6033f6ff4e1b75caa265d2646648b6636c/torch/csrc/jit/ir/alias_analysis.cpp#L86-L145 is a more complete description of the rules... not sure if this is copying the same behavior?

**Review by davidberard98 — APPROVED** [review]
Left some suggestions & some testing ideas

Also - we discussed offline but there's some internal changes you might need to make here in order to add test_schema_info.cpp into an internal TARGETS file.

**Comment by goldenxuett** [comment]
@goldenxuett has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37474862).

**Pull request closed without merging**